src/routes/Profile/Profile.tsx

```diff
@@ -205,20 +207,27 @@ const Profile: React.FunctionComponent = () => {
     }
   }
 
-  function handleOnCopyLinkButtonClick() {
-    // todo: implement logic for fetching the referrer link
+  async function handleOnCopyLinkButtonClick() {
+    if (!wallet.account) {
+      return;
+    }
     setCopyLinkButtonMode('copying');
+    const link = await getReferrerLink(wallet.account);
```

Comment:
Do I understand correctly that this only gets called when they click the link? Could take a few hundred ms to get code from API in the worst case, so is there any value in pre-fetching the code when we load the profile page?

Reply:
I don't want to "create" network calls if not needed. Create them per request of user. Potentially I can add caching if the same "code" is returned always

Reply:
Same code is always returned for a given user.